pages/emotion/[id].js

```diff
@@ -1,19 +1,19 @@
 import EmotionDetails from "@/components/EmotionDetails";
-import { initialEntries } from "@/lib/entries";
 import { useRouter } from "next/router";
 
-export default function EmotionDetailPage() {
+export default function EmotionDetailPage({ entries }) {
   const router = useRouter();
   const { id } = router.query;
 
-  const emotionEntry = initialEntries.find((emotion) => emotion.id === id);
+  const emotionEntry = entries?.find((emotion) => emotion.id === id);
 
   if (!id) {
     return <p>Loading...</p>;
   }
+
   if (!emotionEntry) {
     return <p>No emotion entry found.</p>;
   }
 
-  return <EmotionDetails emotion={emotionEntry} onBack={() => router.back()} />;
+  return <EmotionDetails entry={emotionEntry} onBack={() => router.back()} />;
```

Comment:
You can remove the `onBack` prop:

```suggestion
  return <EmotionDetails entry={emotionEntry} />;
```